### ISSUE TITLE
Revert location extension back to approved PR's

### ISSUE THREAD

**Issue by gmandyam** [w3c/webauthn#800]
Somehow during the recent merges, approved changes to the location extension were eliminated from the current editor's draft (as of Feb. 12):  see https://w3c.github.io/webauthn/#sctn-location-extension and compare to https://pr-preview.s3.amazonaws.com/w3c/webauthn/695/12f2d09...gmandyam:a4fe0fa.html#sctn-location-extension and latest WD https://www.w3.org/TR/webauthn/#sctn-location-extension.

**Comment by selfissued** [comment]
PR #789 intentionally corrected the former misalignment between the W3C Geolocation API and the location extension.  Now they use the same JavaScript representations and an equivalent CBOR representation.  The premise of this issue is faluse.  It needs to be closed with no action taken.

**Comment by gmandyam** [comment]
@selfissued 

Disagree with this argument, and PR 789 should not have addressed such an issue without discussion on the lists.

Previous approved PR added CDDL representation - see https://github.com/w3c/webauthn/pull/695.  PR 789 eliminated it.  This is not consistent with https://github.com/w3c/webauthn/issues/679.  

If there was a concern about inconsistency with the W3C Geoloc Pos object and the loc extension, then why was no issue filed?

**Comment by selfissued** [comment]
There were two serious problems with the previous CBOR for the extension.  First, the required and optional fields were different than those for the W3C Geolocation API Coordinates data structure.  Second, its CBOR representation was an array, whereas the Coordinates data structure is a JavaScript object - not an array.  This means that a custom transformation would be required to support the extension - meaning that it could not be used without explicit browser support for the custom transformation; it could not be used as an unknown extension as described in #738 .  Now that the CBOR and the JavaScript directly correspond, it can.  The changes applied in #789 were part of fixing issue #738 .

You may have not paid addition to it, but there *was* substantial discussion on the list of the representation of the location extension.  See the back-and-forth between @agl , @equalsJeffH , and myself about the representation in the context of #789 .

In short, requiring a custom transformation between the authenticator extension output and the client extension output is the wrong thing to do.  Having a CBOR representation that is syntactically and semantically incompatible with the Coordinates data structure is the wrong thing to do.  Because of the corrections applied, #738 is now fixed.  The proposed reversion would break #738 again.  This must not occur.  Therefore, this issue and PR #801 must be closed with no action.

**Comment by gmandyam** [comment]
There is a misunderstanding as to the source of the data that populates the location extension.  As an example, see the discussion in on whether accuracy should be included in https://github.com/w3c/webauthn/pull/157.  The location extension was not meant to correspond 1-1 with the Geoloc. position object.

**Comment by selfissued** [comment]
The previous location extension definition used the Geolocation API Coordinates data type, but didn't satisfy its requirements (especially by not providing data for a required member).  That would never pass muster in W3C-wide review.  The new text will.  If we want to add a note to implementers about interpretations of and use of the "accuracy" field post-CR, we can do that.

I didn't file a separate PR because this was only one of several extensions that unnecessarily defined custom JavaScript/CBOR transformation procedures.  I fixed them all as part of fixing #738 , which was assigned to me.  PR #765 was available for review for some time and fixed all of these extensions.

@nadalin mentioned that there was a question on today's call about how the JavaScript/CBOR transformations should be done.  Generic transformations were defined by PR #789 and work correctly for this extension, in particular.  See them at end of the section at https://w3c.github.io/webauthn/#extensions .

(@agl - despite our earlier discussion, because the generic transformations for locations are always in the CBOR -> JavaScript direction, they are always correct; no matter what CBOR number type was in the CBOR, it always becomes a double in JavaScript.  We had the particulars of this wrong in our earlier discussion of the handling of this extension.)

**Comment by gmandyam** [comment]
@selfissued 





OK - I see what the problem was in my original proposal.  Thank you very much for the explanation.  We can add the note on accuracy later.

Before I close the PR and this issue, I had a couple of questions:

a) The Geoloc coordinates object is a [NoInterfaceObject] - see https://heycam.github.io/webidl/#NoInterfaceObject.  Can this appear in Client Extension Output?  If not, maybe it should be cast to a different object type.

b) Is your view that CDDL description for this extension is unnecessary because there already exists a JS object (Geoloc API Coordinates) that can be mapped directly to CBOR?  In other words, in this case can it be assumed that the Authenticator Extenion Output must be

CDDL:  locOutput ={latitude: float64, longitude: float64, accuracy: float64}

without explicitly spelling it out?

**Comment by selfissued** [comment]
Thanks for discussing this @gmandyam .

For (a), I don't think the NoInterfaceObject attribute is a problem in this context where it's just being used as a data structure, but others who are experts in JavaScript might want to weigh in.

For (b), yes, CDDL is unnecessary because there's an unambiguous transformation between the JavaScript data structure and the CBOR data structure.

**Comment by gmandyam** [comment]
@selfissued 

Will close.  We may want to consider posting something to https://lists.w3.org/Archives/Public/public-script-coord/ regarding the use of NoInterfaceObjects in Client Extension Output, in order to get expert guidance.
